File: Code/dotNetBasics/ConsoleApp/ADStack/Components/EKFLocalization.cs
using ConsoleApp.ADStack.Models;

namespace ConsoleApp.ADStack.Components
{
    /// <summary>
    /// Loc with Extended Kalman filter
    /// </summary>
    internal class EKFLocalization : IFusion
    {
        public void Bind(IComponent component)
        {
            Console.WriteLine($"[EKFLocalization] Binding component: {component.GetType().Name}");
        }

        public void Start()
        {
            Console.WriteLine("[EKFLocalization] Starting.");
        }

        public void Stop()
        {
            Console.WriteLine("[EKFLocalization] Stopping.");
        }
    }
}

File: Code/Fluent.Java/src/adstack/components/EKFLocalization.java
package adstack.components;

import adstack.models.IComponent;
import adstack.models.IFusion;

/**
 * Loc with Extended Kalman filter
 */
public class EKFLocalization implements IFusion {

	@Override
	public void bind(IComponent component) {
        System.out.println("[EKFLocalization] Binding component: " + component.getClass().getName());
	}

	@Override
	public void start() {
        System.out.println("[EKFLocalization] Starting.");
	}

	@Override
	public void stop() {
        System.out.println("[EKFLocalization] Stopping.");
	}

}
